### PR TITLE
docs: provide top level project documentation

### PR DESCRIPTION
**Pull request by wtrocki**

### CODE REVIEW

**Comment by wtrocki** [comment]
CC @aerogear/charmil-core

File: README.md [hunk 1]
```diff
@@ -13,9 +13,56 @@
...
+## Introduction
 
 
-> NOTE: Project is under construction
+Charmil provides ecosystem of tools to build production ready comand line tools with Cobra Framework.
+
+
+### Architecture
+
+Charmil provides way to build CLI ecosystem where developers can build multiple fragmented CLI's in various repositories
```
Comment (jjkiely) [inline README.md:24]: " a way to build a CLI...."

**Comment by jjkiely** [comment]
This looks awesome :+1: Can't wait to see it be expanded as development continues!

File: README.md [hunk 2]
```diff
@@ -13,9 +13,56 @@
 </p>
 
 
+## Introduction
 
 
-> NOTE: Project is under construction
+Charmil provides ecosystem of tools to build production ready comand line tools with Cobra Framework.
```
Comment (ankithans) [inline README.md:19]: comand typo

File: README.md [hunk 3]
```diff
@@ -13,9 +13,56 @@
...
+
+- Logging
+- Authentication
+- Configuration
+- Commands and Flags standards.
+
+Charmil SDK will typically be introduced into two different CLI
+
+- Host CLI - CLI that is being used by end user that can embed one or more Plugins CLI
+- Plugins CLI - Separate CLI that can be also embeeded into host CLI
```
Comment (ankithans) [inline README.md:36]:
```suggestion
- Plugins CLI - Separate CLI that can be also embedded into host CLI
```

**Review by ankithans — APPROVED** [review]

**Pull request merged**